crux-core/src/crux/api.clj

```diff
@@ -231,8 +234,8 @@
   (submit-tx [this tx-ops]
     (.submitTx this (conform-tx-ops tx-ops)))
 
-  (open-tx-log ^crux.api.ITxLog [this after-tx-id with-ops?]
-    (.openTxLog this after-tx-id with-ops?)))
+  (open-tx-log ^java.io.Closeable [this after-tx-id with-ops?]
```

Comment:
So this PR also affects the tx-log API?

Reply:
yep - same pattern - and I'm aiming to migrate the history API this way too, up next.